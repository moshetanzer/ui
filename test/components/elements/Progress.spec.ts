import { describe, it, expect } from 'vitest'
import { UProgress } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Progress', () => {
  it.each([
    ['basic case', {}],
    ['with custom value', { props: { value: 50 } }],
    ['with indicator', { props: { indicator: true } }],
    ['with slot indicator', { slots: { indicator: () => 'indicator slot' } }],
    ['with custom max', { props: { max: [200] } }],
    ['with steps', { props: { max: [50, 100, 150, 200] } }],
    ['with custom size', { props: { size: 'sm' } }],
    ['with custom color', { props: { color: 'red' } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UProgress.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UProgress)
    expect(html).toMatchSnapshot()
  })
})
