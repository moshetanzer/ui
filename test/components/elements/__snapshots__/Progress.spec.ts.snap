// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Progress > renders basic case correctly 1`] = `
"<div data-v-0f229b8a="" class="w-full flex flex-col gap-2">
  <!--v-if--><progress data-v-0f229b8a="" class="block appearance-none border-none overflow-hidden w-full [&::-webkit-progress-bar]:w-full h-2 rounded-full [&::-webkit-progress-bar]:rounded-full [&::-webkit-progress-bar]:bg-gray-200 [&::-webkit-progress-bar]:dark:bg-gray-700 [@supports(selector(&::-moz-progress-bar))]:bg-gray-200 [@supports(selector(&::-moz-progress-bar))]:dark:bg-gray-700 [&::-webkit-progress-value]:rounded-full [&::-webkit-progress-value]:transition-all [&::-webkit-progress-value]:ease-in-out [&::-moz-progress-bar]:rounded-full text-primary-500 dark:text-primary-400 [&::-webkit-progress-value]:bg-current [&::-moz-progress-bar]:bg-current indeterminate:relative indeterminate:after:rounded-full [&:indeterminate::-webkit-progress-value]:rounded-full [&:indeterminate::-moz-progress-bar]:rounded-full bar-animation-carousel"></progress>
  <!--v-if-->
</div>"
`;

exports[`Progress > renders with custom color correctly 1`] = `
"<div data-v-0f229b8a="" class="w-full flex flex-col gap-2">
  <!--v-if--><progress data-v-0f229b8a="" class="block appearance-none border-none overflow-hidden w-full [&::-webkit-progress-bar]:w-full h-2 rounded-full [&::-webkit-progress-bar]:rounded-full [&::-webkit-progress-bar]:bg-gray-200 [&::-webkit-progress-bar]:dark:bg-gray-700 [@supports(selector(&::-moz-progress-bar))]:bg-gray-200 [@supports(selector(&::-moz-progress-bar))]:dark:bg-gray-700 [&::-webkit-progress-value]:rounded-full [&::-webkit-progress-value]:transition-all [&::-webkit-progress-value]:ease-in-out [&::-moz-progress-bar]:rounded-full text-red-500 dark:text-red-400 [&::-webkit-progress-value]:bg-current [&::-moz-progress-bar]:bg-current indeterminate:relative indeterminate:after:rounded-full [&:indeterminate::-webkit-progress-value]:rounded-full [&:indeterminate::-moz-progress-bar]:rounded-full bar-animation-carousel"></progress>
  <!--v-if-->
</div>"
`;

exports[`Progress > renders with custom max correctly 1`] = `
"<div data-v-0f229b8a="" class="w-full flex flex-col gap-2">
  <!--v-if--><progress data-v-0f229b8a="" class="block appearance-none border-none overflow-hidden w-full [&::-webkit-progress-bar]:w-full h-2 rounded-full [&::-webkit-progress-bar]:rounded-full [&::-webkit-progress-bar]:bg-gray-200 [&::-webkit-progress-bar]:dark:bg-gray-700 [@supports(selector(&::-moz-progress-bar))]:bg-gray-200 [@supports(selector(&::-moz-progress-bar))]:dark:bg-gray-700 [&::-webkit-progress-value]:rounded-full [&::-webkit-progress-value]:transition-all [&::-webkit-progress-value]:ease-in-out [&::-moz-progress-bar]:rounded-full text-primary-500 dark:text-primary-400 [&::-webkit-progress-value]:bg-current [&::-moz-progress-bar]:bg-current indeterminate:relative indeterminate:after:rounded-full [&:indeterminate::-webkit-progress-value]:rounded-full [&:indeterminate::-moz-progress-bar]:rounded-full bar-animation-carousel"></progress>
  <div data-v-0f229b8a="" class="grid grid-cols-1 text-primary-500 dark:text-primary-400 text-sm">
    <div data-v-0f229b8a="" class="transition-all opacity-0 truncate row-start-1 col-start-1 text-end text-gray-500 dark:text-gray-400 opacity-100">200</div>
  </div>
</div>"
`;

exports[`Progress > renders with custom size correctly 1`] = `
"<div data-v-0f229b8a="" class="w-full flex flex-col gap-2">
  <!--v-if--><progress data-v-0f229b8a="" class="block appearance-none border-none overflow-hidden w-full [&::-webkit-progress-bar]:w-full h-1 rounded-full [&::-webkit-progress-bar]:rounded-full [&::-webkit-progress-bar]:bg-gray-200 [&::-webkit-progress-bar]:dark:bg-gray-700 [@supports(selector(&::-moz-progress-bar))]:bg-gray-200 [@supports(selector(&::-moz-progress-bar))]:dark:bg-gray-700 [&::-webkit-progress-value]:rounded-full [&::-webkit-progress-value]:transition-all [&::-webkit-progress-value]:ease-in-out [&::-moz-progress-bar]:rounded-full text-primary-500 dark:text-primary-400 [&::-webkit-progress-value]:bg-current [&::-moz-progress-bar]:bg-current indeterminate:relative indeterminate:after:rounded-full [&:indeterminate::-webkit-progress-value]:rounded-full [&:indeterminate::-moz-progress-bar]:rounded-full bar-animation-carousel"></progress>
  <!--v-if-->
</div>"
`;

exports[`Progress > renders with custom value correctly 1`] = `
"<div data-v-0f229b8a="" class="w-full flex flex-col gap-2">
  <!--v-if--><progress data-v-0f229b8a="" class="block appearance-none border-none overflow-hidden w-full [&::-webkit-progress-bar]:w-full h-2 rounded-full [&::-webkit-progress-bar]:rounded-full [&::-webkit-progress-bar]:bg-gray-200 [&::-webkit-progress-bar]:dark:bg-gray-700 [@supports(selector(&::-moz-progress-bar))]:bg-gray-200 [@supports(selector(&::-moz-progress-bar))]:dark:bg-gray-700 [&::-webkit-progress-value]:rounded-full [&::-webkit-progress-value]:transition-all [&::-webkit-progress-value]:ease-in-out [&::-moz-progress-bar]:rounded-full text-primary-500 dark:text-primary-400 [&::-webkit-progress-value]:bg-current [&::-moz-progress-bar]:bg-current indeterminate:relative indeterminate:after:rounded-full [&:indeterminate::-webkit-progress-value]:rounded-full [&:indeterminate::-moz-progress-bar]:rounded-full" max="100" value="50">50%</progress>
  <!--v-if-->
</div>"
`;

exports[`Progress > renders with indicator correctly 1`] = `
"<div data-v-0f229b8a="" class="w-full flex flex-col gap-2">
  <div data-v-0f229b8a="" class="flex flex-row justify-end min-w-fit transition-all" style="">
    <div data-v-0f229b8a="" class="text-end w-fit text-gray-400 dark:text-gray-500 text-sm">NaN% </div>
  </div><progress data-v-0f229b8a="" class="block appearance-none border-none overflow-hidden w-full [&::-webkit-progress-bar]:w-full h-2 rounded-full [&::-webkit-progress-bar]:rounded-full [&::-webkit-progress-bar]:bg-gray-200 [&::-webkit-progress-bar]:dark:bg-gray-700 [@supports(selector(&::-moz-progress-bar))]:bg-gray-200 [@supports(selector(&::-moz-progress-bar))]:dark:bg-gray-700 [&::-webkit-progress-value]:rounded-full [&::-webkit-progress-value]:transition-all [&::-webkit-progress-value]:ease-in-out [&::-moz-progress-bar]:rounded-full text-primary-500 dark:text-primary-400 [&::-webkit-progress-value]:bg-current [&::-moz-progress-bar]:bg-current indeterminate:relative indeterminate:after:rounded-full [&:indeterminate::-webkit-progress-value]:rounded-full [&:indeterminate::-moz-progress-bar]:rounded-full bar-animation-carousel"></progress>
  <!--v-if-->
</div>"
`;

exports[`Progress > renders with slot indicator correctly 1`] = `
"<div data-v-0f229b8a="" class="w-full flex flex-col gap-2">indicator slot<progress data-v-0f229b8a="" class="block appearance-none border-none overflow-hidden w-full [&::-webkit-progress-bar]:w-full h-2 rounded-full [&::-webkit-progress-bar]:rounded-full [&::-webkit-progress-bar]:bg-gray-200 [&::-webkit-progress-bar]:dark:bg-gray-700 [@supports(selector(&::-moz-progress-bar))]:bg-gray-200 [@supports(selector(&::-moz-progress-bar))]:dark:bg-gray-700 [&::-webkit-progress-value]:rounded-full [&::-webkit-progress-value]:transition-all [&::-webkit-progress-value]:ease-in-out [&::-moz-progress-bar]:rounded-full text-primary-500 dark:text-primary-400 [&::-webkit-progress-value]:bg-current [&::-moz-progress-bar]:bg-current indeterminate:relative indeterminate:after:rounded-full [&:indeterminate::-webkit-progress-value]:rounded-full [&:indeterminate::-moz-progress-bar]:rounded-full bar-animation-carousel"></progress>
  <!--v-if-->
</div>"
`;

exports[`Progress > renders with steps correctly 1`] = `
"<div data-v-0f229b8a="" class="w-full flex flex-col gap-2">
  <!--v-if--><progress data-v-0f229b8a="" class="block appearance-none border-none overflow-hidden w-full [&::-webkit-progress-bar]:w-full h-2 rounded-full [&::-webkit-progress-bar]:rounded-full [&::-webkit-progress-bar]:bg-gray-200 [&::-webkit-progress-bar]:dark:bg-gray-700 [@supports(selector(&::-moz-progress-bar))]:bg-gray-200 [@supports(selector(&::-moz-progress-bar))]:dark:bg-gray-700 [&::-webkit-progress-value]:rounded-full [&::-webkit-progress-value]:transition-all [&::-webkit-progress-value]:ease-in-out [&::-moz-progress-bar]:rounded-full text-primary-500 dark:text-primary-400 [&::-webkit-progress-value]:bg-current [&::-moz-progress-bar]:bg-current indeterminate:relative indeterminate:after:rounded-full [&:indeterminate::-webkit-progress-value]:rounded-full [&:indeterminate::-moz-progress-bar]:rounded-full bar-animation-carousel"></progress>
  <div data-v-0f229b8a="" class="grid grid-cols-1 text-primary-500 dark:text-primary-400 text-sm">
    <div data-v-0f229b8a="" class="transition-all opacity-0 truncate row-start-1 col-start-1 text-end text-gray-500 dark:text-gray-400 opacity-100">50</div>
    <div data-v-0f229b8a="" class="transition-all opacity-0 truncate row-start-1 col-start-1 text-end">100</div>
    <div data-v-0f229b8a="" class="transition-all opacity-0 truncate row-start-1 col-start-1 text-end">150</div>
    <div data-v-0f229b8a="" class="transition-all opacity-0 truncate row-start-1 col-start-1 text-end">200</div>
  </div>
</div>"
`;
